src/main/java/org/avaje/agentloader/AgentLoader.java

```diff
@@ -151,6 +175,68 @@ private static final boolean isWindows() {
     return File.separatorChar == '\\';
   }
 
+  /**
+   * This method takes the jar:file:path-to-filename.war!/WEB-INF/jar/jar-file/ offset that is included in the
+   * url classpath and extracts out a single jar containing the files in that match that url.
+   *
+   * @param path    - full url entry in the classpath
+   * @param agentName - the agent name that we are trying to match
+   * @return null if it fails or a full path to the jar file if it succeeds
+   */
+  public static String extractJar(URL path, String agentName) {
+    String fullPath = null;
+
+    String[] jarNames = path.getPath().split(":");
+
+    if (jarNames.length >= 2) {
```

Comment:
Thanks for your comments. BTW, do you think it is OK for me to switch the logger to slf4j instead of logging?